https://duckdb-node.s3.amazonaws.com/duckdb-v0.6.1-node-v115-darwin-arm64.tar.gz  returns access denied

### What happens?

Note this looks similar to the issue at: https://github.com/duckdb/duckdb/issues/1903

I'm trying to install the DuckDB SQL tools for VSCode (M2 Macbook)
https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.duckdb-sql-tools

It looks like there is a dependency on duckdb-async@0.6.1, it looks like VS Code is executing `npm install duckdb-async@0.6.1 && exit 0` to install, but it errors out. Full log is pasted below, but it seems that the key error is 

```shell
npm ERR! node-pre-gyp ERR! install response status 403 Forbidden on https://duckdb-node.s3.amazonaws.com/duckdb-v0.6.1-node-v115-darwin-arm64.tar.gz 
``` 

Then it appears to try to compile from source, which also fails. See full log below:

```shell
npm install duckdb-async@0.6.1 && exit 0
(base) ryan@RYANs-MBP vscode-sqltools % npm install duckdb-async@0.6.1 && exit 0
conda activate py311 ⠧ idealTree:vscode-sqltools: sill idealTree buildDeps
npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm ERR! code 1
npm ERR! path /Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build
npm ERR! CXX(target) Release/obj.target/duckdb/src/duckdb_node.o
npm ERR! Failed to execute '/opt/homebrew/Cellar/node/20.0.0/bin/node /opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding/duckdb.node --module_name=duckdb --module_path=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v115' (1)
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
npm ERR! node-pre-gyp info using node@20.0.0 | darwin | arm64
npm ERR! node-pre-gyp info check checked for "/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding/duckdb.node" (not found)
npm ERR! node-pre-gyp http GET https://duckdb-node.s3.amazonaws.com/duckdb-v0.6.1-node-v115-darwin-arm64.tar.gz
npm ERR! node-pre-gyp ERR! install response status 403 Forbidden on https://duckdb-node.s3.amazonaws.com/duckdb-v0.6.1-node-v115-darwin-arm64.tar.gz 
npm ERR! node-pre-gyp WARN Pre-built binaries not installable for duckdb@0.6.1 and node@20.0.0 (node-v115 ABI, unknown) (falling back to source compile with node-gyp) 
npm ERR! node-pre-gyp WARN Hit error response status 403 Forbidden on https://duckdb-node.s3.amazonaws.com/duckdb-v0.6.1-node-v115-darwin-arm64.tar.gz 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@20.0.0 | darwin | arm64
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@20.0.0 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.9.6 found at "/Library/Developer/CommandLineTools/usr/bin/python3"
npm ERR! gyp info spawn /Library/Developer/CommandLineTools/usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/ryan/Library/Caches/node-gyp/20.0.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/ryan/Library/Caches/node-gyp/20.0.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/ryan/Library/Caches/node-gyp/20.0.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@20.0.0 | darwin | arm64
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! clang: error: no such file or directory: 'Support/vscode-sqltools/node_modules/node-addon-api'
npm ERR! make: *** [Release/obj.target/duckdb/src/duckdb_node.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:203:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:511:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm ERR! gyp ERR! System Darwin 22.4.0
npm ERR! gyp ERR! command "/opt/homebrew/Cellar/node/20.0.0/bin/node" "/opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding/duckdb.node" "--module_name=duckdb" "--module_path=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v115"
npm ERR! gyp ERR! cwd /Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb
npm ERR! gyp ERR! node -v v20.0.0
npm ERR! gyp ERR! node-gyp -v v9.3.1
npm ERR! gyp ERR! not ok 
npm ERR! node-pre-gyp ERR! build error 
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/opt/homebrew/Cellar/node/20.0.0/bin/node /opt/homebrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding/duckdb.node --module_name=duckdb --module_path=/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb/lib/binding --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v115' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:511:28)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1098:16)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:304:5)
npm ERR! node-pre-gyp ERR! System Darwin 22.4.0
npm ERR! node-pre-gyp ERR! command "/opt/homebrew/Cellar/node/20.0.0/bin/node" "/Users/ryan/Library/Application Support/vscode-sqltools/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
npm ERR! node-pre-gyp ERR! cwd /Users/ryan/Library/Application Support/vscode-sqltools/node_modules/duckdb
npm ERR! node-pre-gyp ERR! node -v v20.0.0
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /Users/ryan/.npm/_logs/2023-04-28T09_12_55_612Z-debug-0.log
```



### To Reproduce

Try to install the DuckDB SQL tools for VSCode (M2 Macbook)
https://marketplace.visualstudio.com/items?itemName=RandomFractalsInc.duckdb-sql-tools

### OS:

MacOS M2

### DuckDB Version:

v0.7.1 b00b93f0b1

### DuckDB Client:

DuckDB SQL tools for VSCode

### Full Name:

Ryan Rozich

### Affiliation:

Oakmont Partners, LLC

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

--- a/.github/workflows/NodeJS.yml
+++ b/.github/workflows/NodeJS.yml
@@ -65,7 +65,7 @@ jobs:
       DUCKDB_NODE_BUILD_CACHE: 0
     strategy:
       matrix:
-        node: [ '10', '12', '14', '15', '16', '17', '18', '19' ]
+        node: [ '10', '12', '14', '15', '16', '17', '18', '19', '20' ]
         target_arch: [ x64, arm64 ]
         isRelease:
           - ${{ startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/master' }}
@@ -89,6 +89,9 @@ jobs:
             node: 18
             target_arch: x64
           - isRelease: false
+            node: 19
+            target_arch: x64
+          - isRelease: false
             node: 12
             target_arch: arm64
           - isRelease: false
@@ -105,6 +108,9 @@ jobs:
             target_arch: arm64
           - isRelease: false
             node: 18
+            target_arch: arm64
+          - isRelease: false
+            node: 19
             target_arch: arm64
 
     steps:
@@ -164,7 +170,7 @@ jobs:
     strategy:
       matrix:
         target_arch: [ x64, arm64 ]
-        node: [ '10', '12', '14', '15', '16', '17', '18', '19' ]
+        node: [ '10', '12', '14', '15', '16', '17', '18', '19', '20' ]
         isRelease:
           - ${{ startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/master' }}
         exclude:
@@ -240,7 +246,7 @@ jobs:
 
     strategy:
       matrix:
-        node: [ '10', '12', '14', '15', '16', '17', '18', '19' ]
+        node: [ '10', '12', '14', '15', '16', '17', '18', '19', '20' ]
         isRelease:
           - ${{ startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/master' }}
         exclude:
@@ -256,6 +262,8 @@ jobs:
             node: 17
           - isRelease: false
             node: 18
+          - isRelease: false
+            node: 19
 
     steps:
       - uses: actions/setup-python@v4

